### PR TITLE
Make the illuminate driver threadsafe

### PR DESCRIPTION
**Pull request by vetonsaliu8**
Add re-entrant locks to methods that communicate with the illumination hardware

### CODE REVIEW

File: pyilluminate/illuminate.py [hunk 1]
```diff
@@ -130,6 +180,7 @@ def find(serial_numbers=None):
         return devices
 
     @staticmethod
+    @with_thread_lock_static
```
Comment (hmaarrfk) [inline pyilluminate/illuminate.py]: do we need this? is any communication opened here?

File: pyilluminate/illuminate.py [hunk 2]
```diff
@@ -53,6 +58,51 @@
 # https://forum.pjrc.com/threads/54773-Inreasing-USB-Serial-Buffer-Teensy-3-2
 MAX_ARGUMENT_CHAR_COUNT = 64 * 1
 
+thread_lock = RLock()
```
Comment (hmaarrfk) [inline pyilluminate/illuminate.py]: This won't work with multiple different LED boards at the same time.

I don't think we need to lock 'static" functions. They don't use the serial port itself. They just ask the operating system for the status of the USB port.

**Review by hmaarrfk — CHANGES_REQUESTED** [review]
we don't need the static lock.

The second problem is that the current implementation does not enable two different LED boards to be used at the same time. 

I don't think that this is OK.

We should allow two boards to be used at the same time. The operating system will take care of the USB contention.

**Comment by vetonsaliu8** [comment]
Fixed that and other small issues.  Also added locks to fake_illuminate.

**Comment by hmaarrfk** [comment]
Can you have a look at automated tests and address the failures reported please

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,7 @@
+## 0.6.15 (2022/04/20)
```
Comment (hmaarrfk) [inline CHANGELOG.md]:
```suggestion
## 0.6.15 (2022/04/21)
```

File: pyilluminate/illuminate.py [hunk 4]
```diff
@@ -94,7 +130,8 @@ class Illuminate:
         '_has_autoupdate',
         '_autoclear',
         '_autoupdate',
-        '_scale_factor'
+        '_scale_factor',
+        '_thread_lock'
```
Comment (hmaarrfk) [inline pyilluminate/illuminate.py]:
```suggestion
        '_thread_lock',
```
This makes the diff smaller latter

**Comment by hmaarrfk** [comment]
Thanks veton!

**Pull request merged**